aviatrix/resource_aviatrix_saml_endpoint.go

```diff
@@ -29,15 +29,24 @@ func resourceAviatrixSamlEndpoint() *schema.Resource {
 				Description: "SAML Endpoint Name.",
 			},
 			"idp_metadata_type": {
-				Type:        schema.TypeString,
-				Required:    true,
-				Description: "Type of IDP Metadata.",
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  "Type of IDP Metadata.",
+				ValidateFunc: validation.StringInSlice([]string{"URL", "Text"}, false),
 			},
 			"idp_metadata": {
 				Type:        schema.TypeString,
-				Required:    true,
+				Optional:    true,
+				Default:     "",
 				Description: "IDP Metadata.",
 			},
+			"idp_metadata_url": {
```

Comment:
```
	// ConflictsWith is a set of schema keys that conflict with this schema.
	// This will only check that they're set in the _config_. This will not
	// raise an error for a malfunctioning resource that sets a conflicting
	// key.
	ConflictsWith []string
```
You can define that idp_metadata ConflictsWith idp_metadata_url. So at _plan_ time you get an error.

Reply:
AtLeastOneOf also looks like it would apply to idp_metadata and idp_metadata_url
```
	// AtLeastOneOf is a set of schema keys that, when set, at least one of
	// the keys in that list must be specified.
```

Reply:
Looks like AtLeastOneOf is a better one as one or the other is required

Reply:
Actually both need to be used